src/wp-includes/post.php

```diff
@@ -6038,192 +6031,57 @@ function get_pages( $args = array() ) {
 		return false;
 	}
 
-	// $args can be whatever, only use the args defined in defaults to compute the key.
-	$key          = md5( serialize( wp_array_slice_assoc( $parsed_args, array_keys( $defaults ) ) ) );
-	$last_changed = wp_cache_get_last_changed( 'posts' );
-
-	$cache_key = "get_pages:$key:$last_changed";
-	$cache     = wp_cache_get( $cache_key, 'posts' );
-	if ( false !== $cache ) {
-		_prime_post_caches( $cache, false, false );
-
-		// Convert to WP_Post instances.
-		$pages = array_map( 'get_post', $cache );
-		/** This filter is documented in wp-includes/post.php */
-		$pages = apply_filters( 'get_pages', $pages, $parsed_args );
```

Comment:
maybe discussed already: is it ok to remove this filter application? do any plugins use this?

Reply:
This is retained below (after the call to WP_Query) https://github.com/WordPress/wordpress-develop/pull/3178/files#diff-ef9c5475561d459917dac57d21e08a6dfeb36f8b1c44b978d3696aec8b97c405R6115